zkml/docs/src/llms-layers/qkv.md

```diff
@@ -0,0 +1,95 @@
+# QKV
+
+## Description of the Layer
+The **QKV** layer is the first layer found in each encoder/decoder of an LLM architecture, and it is employed to produce, for each word (or better, *token*) being processed by the LLM three vectors:
+
+- A query vector `q`
+- A key vector `k`
+- A value vector `v`
+
+These vectors are all computed from a vector `x` representing a token processed by the LLM. More specifically, each of this vector is obtained by multiplying `x` with corresponding weight matrices  $W_q$, $W_k$ and $W_v$, and then each resulting vector is added to a corresponding bias vector $B_q$, $B_k$ and $B_v$. Since an LLM usually needs to process more than 1 token at the same time, the computation of `q`, `k` and `v` vectors for all the input tokens is done in a "batched" fashon over matrices rather than over single vectors:
```

Comment:
Can you write the equation instead / or on top of describing them with long sentences ? Would be much more succinct !

Reply:
Done in commit https://github.com/Lagrange-Labs/deep-prove/pull/123/commits/a8f86fba6856eed02b5775488b89e187fde4c33b